rest/src/main/java/org/jboss/pnc/rest/notifications/model/NotificationFactory.java

```diff
@@ -0,0 +1,34 @@
+package org.jboss.pnc.rest.notifications.model;
+
+import org.jboss.pnc.spi.BuildStatus;
+import org.jboss.pnc.spi.events.BuildStatusChangedEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+import java.util.HashMap;
+import java.util.Map;
+
+@ApplicationScoped
+public class NotificationFactory {
+
+    private Map<BuildStatus, NotificationEventType> externalEvents = new HashMap<>();
+
+    public NotificationFactory() {
+        externalEvents.put(BuildStatus.REPO_SETTING_UP, NotificationEventType.BUILD_STARTED);
```

Comment:
Yes, we can add a mapping for that if you wish (but on the other hand we 
don't have any requirements for this at the moment, so I'm not entirely 
convinced that we should) but as I mentioned before, I would like to 
preserve this safety buffer between Backend and UI.

On 07/02/2015 10:07 AM, Matej Lazar wrote:
>
> In 
> rest/src/main/java/org/jboss/pnc/rest/notifications/model/NotificationFactory.java 
> <https://github.com/project-ncl/pnc/pull/425#discussion_r33755075>:
>
> > +package org.jboss.pnc.rest.notifications.model;
> > +
> > +import org.jboss.pnc.spi.BuildStatus;
> > +import org.jboss.pnc.spi.events.BuildStatusChangedEvent;
> > +
> > +import javax.enterprise.context.ApplicationScoped;
> > +import java.util.HashMap;
> > +import java.util.Map;
> > +
> > +@ApplicationScoped
> > +public class NotificationFactory {
> > +
> > +    private Map<BuildStatus, NotificationEventType> externalEvents = new HashMap<>();
> > +
> > +    public NotificationFactory() {
> > +        externalEvents.put(BuildStatus.REPO_SETTING_UP, NotificationEventType.BUILD_STARTED);
>
> Some other "clients" might want so show more progress details. I find 
> it useful to see that my task is eg. COLLECTING_RESULTS_* or that task 
> failed due to BUILD_ENV_SETUP and not because of the 
> BUILD_COMPLETED_WITH_ERROR. Or maybe "WAITING_FOR_DEPENDENCIES" tells 
> that the particular component is still waiting to be scheduled. Don't 
> you think we should show this to the users ?
> And org.jboss.pnc.spi.BuildStatus is already a part of the SPI.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/project-ncl/pnc/pull/425/files#r33755075>.
>